docs: Document migration scope (#772)
UI5/linter: docs/Migration-Limitations.md

> # Autofix Migration Limitations

Reviewer: Is it ok to use the term `Limitation` in this context? CC: @KlattG

Reviewer 2: AFAIR from a discussion in the UA community some years back, the term 'Limitation' has serious legal implication, implying that promised functionality is not available after all. In the Developer's Guide we therefore talk of 'Restrictions'  whenever things do not work the way users might expect.

Reviewer 2: See also https://one.int.sap/company/management_systems/limitations

Reviewer: I would propose "Restrictions" or "Scope". Also, should we replace the term "migration" with "autofix" in the name of the document? E.g. "Autofix Scope"

Reviewer 2: We had this discussion for the dev guide recently - the upshot was that 'migration' would be inappropriate for new users, ie those not upgrading from 1.x 
So "Scope of Autofix (Feature)" might be preferable then.

Reviewer 3: Done with [ccf9653](https://github.com/UI5/linter/pull/772/commits/ccf9653a262fac3c0c439f5e717fc0039fc72abb) ?

> This document lists APIs that are not migrated or can't be migrated automatically by the UI5 linter. These APIs require manual migration or special handling.

Reviewer: ```suggestion
This document lists APIs that are not migrated or can't be migrated automatically by the UI5 linter at this time. These APIs require manual migration or special handling.
```

> **Note:** This list is not exhaustive; there are more APIs that are currently not migrated automatically. This document provides examples of common APIs that require manual migration.

- [Autofix Migration Limitations](#autofix-migration-limitations)
	- [General Limitations](#general-limitations)
	- [jQuery.sap APIs](#jquerysap-apis)
	- [jQuery Plugins](#jquery-plugins)
	- [Global API Usage](#global-api-usage)
	- [Module Imports](#module-imports)
	- [Core APIs (See #619)](#core-apis-see-619)
		- [Template APIs](#template-apis)
		- [Event Handling APIs](#event-handling-apis)
		- [Error Handling APIs](#error-handling-apis)
		- [Rendering APIs](#rendering-apis)
		- [Model APIs](#model-apis)
		- [Component/Application APIs](#componentapplication-apis)
		- [Other Core APIs](#other-core-apis)
	- [Core Configuration APIs (See #620)](#core-configuration-apis-see-620)
	- [Sync to Async API Changes](#sync-to-async-api-changes)
		- [Library Loading](#library-loading)
		- [Component Creation](#component-creation)
		- [Resource Bundle Loading](#resource-bundle-loading)
		- [Other Async APIs](#other-async-apis)

## General Limitations

> - **Change from sync to async APIs** - Requires restructuring code flow, which cannot be done automatically

Reviewer: ```suggestion
- **Change from sync to async APIs** - Requires restructuring code flow (often affecting multiple files), which cannot be done automatically
```

> - **Complex API replacements** - Some APIs require complex replacements with multiple lines of code

Reviewer: ```suggestion
- **Complex API replacements** - Some APIs require complex replacements with multiple API calls and new local variables, this is currently not supported
```

- [**jQuery.sap.registerResourcePath**](https://ui5.sap.com/1.136/#/api/jQuery.sap%23methods/jQuery.sap.registerResourcePath) - Currently not implemented, see [#588](https://github.com/UI5/linter/issues/588)
- [**jQuery.sap.removeUrlWhitelist**](https://ui5.sap.com/1.136/#/api/jQuery.sap%23methods/jQuery.sap.removeUrlWhitelist) - Too complex, see [#657](https://github.com/UI5/linter/issues/657#issuecomment-2983899399)

## jQuery Plugins

- [**jQuery#control**](https://ui5.sap.com/1.136/#/api/jQuery%23methods/control) - No direct replacement, see [#578](https://github.com/UI5/linter/issues/578#issuecomment-2866034841)

## Global API Usage

- **Assignments to global variables** - Cannot be migrated automatically
- **`delete` expressions on globals** - Cannot be migrated automatically, see [#668](https://github.com/UI5/linter/issues/668)

## Module Imports

- **Pseudo module imports** - Currently not implemented, see [#715](https://github.com/UI5/linter/issues/715)

## Core APIs (See [#619](https://github.com/UI5/linter/issues/619))

> Many methods on the Core API (`sap/ui/core/Core` / `sap.ui.getCore()`) cannot be migrated automatically:

Reviewer: ```suggestion
Many methods on the Core API (accessed via either the `sap/ui/core/Core` module import or via `sap.ui.getCore()`) cannot be migrated automatically:
```

- [**Core#lock**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/lock) - Concept has been discarded
- [**Core#registerPlugin**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/registerPlugin) - Concept has been discarded
- [**Core#setRoot**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/setRoot) - No direct replacement
- [**Core#setThemeRoot**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/setThemeRoot) - No direct replacement
- [**Core#unlock**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/unlock) - Concept has been discarded
- [**Core#unregisterPlugin**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/unregisterPlugin) - Concept has been discarded
- [**Core.extend**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/sap.ui.core.Core.extend) - No direct replacement
- [**Core.getMetadata**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/sap.ui.core.Core.getMetadata) - No direct replacement

## Core Configuration APIs (See [#620](https://github.com/UI5/linter/issues/620))

> Some methods on the Configuration API (`sap.ui.getCore().getConfiguration()` / `sap/ui/core/Configuration`) cannot be migrated automatically:

Reviewer: ```suggestion
Some methods on the Configuration API (accessed via either the  `sap/ui/core/Configuration` module import or via `sap.ui.getCore().getConfiguration()`) cannot be migrated automatically:
```

- [**Configuration.getManifestFirst**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Configuration%23methods/sap.ui.core.Configuration.getManifestFirst) - No replacement
- [**Configuration.getNoDuplicateIds**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Configuration%23methods/sap.ui.core.Configuration.getNoDuplicateIds) - Concept has been discarded
- [**Configuration.getOriginInfo**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Configuration%23methods/sap.ui.core.Configuration.getOriginInfo) - Concept has been discarded
- [**Configuration.getRootComponent**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Configuration%23methods/sap.ui.core.Configuration.getRootComponent) - Concept has been discarded
- [**Configuration.getStatisticsEnabled**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Configuration%23methods/sap.ui.core.Configuration.getStatisticsEnabled) - No replacement
- [**Configuration.getVersion**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Configuration%23methods/sap.ui.core.Configuration.getVersion) - Sync to async replacement

## Sync to Async API Changes

> The UI5 linter cannot automatically migrate APIs that change from synchronous to asynchronous. This is a general limitation as it would require restructuring the code flow. Examples include:

Reviewer: ```suggestion
Currently, UI5 linter cannot automatically migrate APIs that execute synchronously if their replacement would execute asynchronously (i.e. returns a promise). This is a general limitation as it would require restructuring the code flow. Examples include:
```

### Library Loading

- [**Core#loadLibrary**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/loadLibrary)
  - Only migrated to `sap/ui/core/Lib.load()` if `async: true` is explicitly specified
  - Synchronous library loading cannot be automatically migrated

### Component Creation

- [**Core#createComponent**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/createComponent)
  - Only migrated to `sap/ui/core/Component.create()` if `async: true` is explicitly specified
  - Synchronous component creation cannot be automatically migrated

### Resource Bundle Loading

- [**Core#getLibraryResourceBundle**](https://ui5.sap.com/1.136/#/api/sap.ui.core.Core%23methods/getLibraryResourceBundle)
  - Not migrated if any argument is a boolean with value true (which would make it return a promise)
  - Synchronous resource bundle loading cannot be automatically migrated to asynchronous

> ### Other Async APIs

Reviewer: ```suggestion
### Other APIs
```
